### DISCUSSION TITLE
Unable to Identify Old Objects

### DISCUSSION THREAD

**Discussion by nkupate1** [thoughtspot/cs_tools#152] [Q&A] [answered]
Hello Everyone,
I am trying to run command (PFA) using cs_tools to identify the objects that are not in use or haven't modified since 120 days. 

However, I am getting below message every time as “No Worksheet Found” even though I change the days to 180 or more and I am pretty sure there are Liveboards and answers haven’t modified since last 90 days. 

Can you please help to identify the objects older than 120 days.

![image](https://github.com/thoughtspot/cs_tools/assets/153696411/09349ce9-2b51-42c4-a92e-a4d12af9a9bd)

**Comment by boonhapus** [comment, marked as answer]
Hey @nkupate1 -- Archiver uses the TS: BI Server Worksheet internally to determine whether or not certain objects have been recently searched against (TS: BI is an activity / history table). The error here is saying we can't find a worksheet with that name which would suggest to me one of two things..

1/ The user you defined in your `test` config is not an admin or isn't able to see TS: BI Server
2/ You have Orgs enabled on your cluster, and the last org this user was associated with is not the Primary org.

Do either of these things sound familiar?

**Reply by nkupate1** [reply]
Thanks a Lot boonhapus. It's working now when I gave admin rights to cs_tools user.

**Comment by boonhapus** [comment]
Sidenote: it looks like you might be running an older version of CS Tools - there should be some log lines at the top of the command execution stating what your login mechanism is, and if you're authenticating towards a particular org.

You might try running `powershell -c "IRM https://raw.githubusercontent.com/thoughtspot/cs_tools/master/cs_tools/updater/_bootstrapper.py | python - --install"` to upgrade to `v1.5.8` . This likely won't solve your problem.. but it will get you to a more supported version! 😄
